src/backend/optimizer/plan/aqumv.c

```diff
@@ -168,12 +169,18 @@ answer_query_using_materialized_views(PlannerInfo *root,
 		matviewRel = table_open(rewrite_tup->ev_class, AccessShareLock);
 		need_close = true;
 
+		if (!RelationIsPopulated(matviewRel))
+			continue;
+
 		/*
 		 * AQUMV
 		 * Currently the data of IVM is always up-to-date if there were.
-		 * Take care of this when IVM defered-fefresh is supported.
+		 * Take care of this when IVM defered-fefresh is supported(in SERVERLESS mode).
+		 * 
+		 * Normal materialized views could also be used if its data is up to date.
 		 */
-		if (!(RelationIsIVM(matviewRel) && RelationIsPopulated(matviewRel)))
+		if (!RelationIsIVM(matviewRel) &&
+			!MatviewIsGeneralyUpToDate(RelationGetRelid(matviewRel)))
```

Comment:
For external table, should we use a GUC to control whether to replace query by MV.

Reply:
Yes, it's on the todo-list which has not been implemented.
We search gp_matview_aux to find candidate views, currently only normal base tables could be recorded in gp_matview_aux.
Will do that when we support complex query considering external/foreign table.

Reply:
Got it. Let's merge this PR first.